src/toastify.ts

```diff
@@ -0,0 +1,43 @@
+import { Toast } from 'react-toastify'
```

Comment:
Why did you need to create this component?

I've got a bit lost in the code below. I get that you override/wrap the methods of toast, probably trying to lazy load it.

My understanding is you only load it's css and the `toast` export (I understand that's the biggest part of it), when you use one of it's methods. Also you delay the configuration of it.

am I getting it somewhat right?

Reply:
Yes
So `setupToastify.ts` which includes the library and css synchronously is like that to be put in one chunk
Then in `toastify.ts` it is imported dynamically and reexports a sort of proxy that wraps every method and makes it a promise. That way the interface stays the same, but every method is promisified. `setupToastify` code is run on the first invocation of `toast.<any_method>`, then cached thanks to how dynamic import works.
Thus where we actually need to use it we have:
```
import { toast } from 'toastify'
// ...
toast.success(`The token ${tokenDisplayName} has been enabled for trading`)
```


An alternative would be:
```ts
import('setupToastify').then(({toast}) => toast.success(`The token ${tokenDisplayName} has been enabled for trading`))
```
every time we would need a toast :clinking_glasses: